src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -50,6 +51,12 @@
 register_cli_argument('resource create', 'is_full_object', action='store_true',
                       help='Indicates that the properties object includes other options such as location, tags, sku, and/or plan.')
 
+register_cli_argument('resource lock', 'resource_group', resource_group_name_type)
+register_cli_argument('resource lock', 'resource_name', options_list=('--resource-name'), validator=validate_resource_lock)
+register_cli_argument('resource lock', 'ids', nargs='+', options_list=('--ids'), help='One or more resource IDs (space delimited). If provided, no other "Resource Id" arguments should be specified.')
+register_cli_argument('resource lock', 'lock_name', options_list=('--name', '-n'), help='Name of the lock')
+register_cli_argument('resource lock', 'level', options_list=('--lock-type', '-t'), **enum_choice_list([LockLevel.can_not_delete, LockLevel.read_only]))
```

Comment:
you can consolidate with `group` and `account` to avoid duplicates, e.g. https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py#L173-L177